### PR TITLE
Baseline Jpegs can contain more than one scan

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
@antonfirsov Well the first part was an easy fix. We were stopping processing after reading the initial SOS (Start Of Scan) marker because is was a baseline jpeg. Baseline Jpegs, while generally have only one scan, can actually have more than one. 

That exposed an issue we were having with missing EOF (End of File) markers on baseline files. I was able to copy the behavior from PdfJS port though to counter that.

I'd really like to try and work with you post V1 when we have time to try to bring the best of both decoders together into a better one. I think we have more overhead than we need using the InputProcessor for anything other than spectral decoding within the SOS segment. The byte stuffing and reversal should only be needed there.

### CODE REVIEW

File: src/ImageSharp/Formats/Jpeg/GolangPort/Components/Decoder/InputProcessor.cs [hunk 1]
```diff
@@ -107,6 +107,13 @@ public byte ReadByte()
             return this.Bytes.ReadByte(this.InputStream);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public OrigDecoderErrorCode ReadByteUnsafe(out byte result)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Jpeg/GolangPort/Components/Decoder/InputProcessor.cs]: Ah that's better! 👍

Comment (antonfirsov) [reply src/ImageSharp/Formats/Jpeg/GolangPort/Components/Decoder/InputProcessor.cs]: Actually defaulting to this logic would be the best, but I felt it's more safe to continuously refactor stuff towards it, also keeping the exception throwing where we haven't yet implemented the "smart" error management.

**Comment by JimBobSquarePants** [comment]
Glad it wasn't something buried deep in the scan decoder or something else crazy. You're added method is much neater. 

I'll merge once the build is passed, hopefully tweaking the tests will bring up the coverage.

**Comment by antonfirsov** [comment]
@JimBobSquarePants  re mixing the two decoders, this is how I imagine it:
1. Adapt `OrigJpegScanDecoder` into the PdfJs stream parsing logic, this can be tested & debugged independently with the help of `SpectralJpegTests`
2. Implement `IRawJpegData` by the pdfjs port, so we can feed the spectral results to `JpegImagePostProcessor`

**Pull request merged**